### ISSUE TITLE
Office 2016 data showed only with /force option

### ISSUE THREAD

**Issue by alexchini** [PluginsOCSInventory-NG/officepack#38]
Hi i ve an issue with the plugin the data about office 16 are showed only if i lunch the OCSInventory.exe /force from command prompt in a windows client. All the versions server client plugin are the last i found. Thanks

**Comment by gehasia** [comment]
I upvote this one, plugin is unusable as this !

**Comment by etherforea** [comment]
+1

**Comment by helicrenz** [comment]
Hello @alexchini,

May i know what office plugin did you install? mine doesnt have any data may you please help

**Comment by alexchini** [comment]
Hi @helicrenz 

I'm using the last version available the 2.0 of 09/2017  with ocs 2.5 but i had the same problems with the ocs 2.3.

The plugin for my case work only if you start the ocs client on your pc  manually  with OCSInventory.exe 
 /force don't work if you start OCSInventory.exe  as a windows service. 

Keep in mind that ocs client use a vb script for parse the registy keys but unfortunally microsoft  write the registry keys of the office versions in different positions and with different logic so  some version of office are not find by ocs.

**Comment by helicrenz** [comment]
hello @alexchini,

I tried to install https://github.com/PluginsOCSInventory-NG/officepack/releases/tag/2.0. But im getting error.

![image](https://user-images.githubusercontent.com/42835226/46150965-5ca9fc00-c2a0-11e8-96c6-c0e6214d26c8.png)

its working other plugin but i cant manage to make it work 2.0. any idea?

**Comment by alexchini** [comment]
ah ur problem is on the server side, maybe the ocs log can give some lead. 

As first idea, if u don't have done, check the files owner of the zip files in the download directory of the server. they must be apache or www-data this depend on the OS u r using.

**Comment by helicrenz** [comment]
@alexchini this plug-in work for me https://github.com/PluginsOCSInventory-NG/officepack/files/1309943/officepack.zip but the thing is it wont show the office key and version. When checking the databased. i see the tables getting the data.



for office pack 2.0 https://github.com/PluginsOCSInventory-NG/officepack/releases.  This want doesnt work im getting error like the picture above. I check the logs. 

Here is the output
tail -f /var/log/apache/error.log
tail -f /var/log/apache2/error.log
[Thu Sep 27 15:32:38.101368 2018] [php7:warn] [pid 2730] [client 10.234.32.6:65164] PHP Warning:  mysqli_query(): Empty query in /usr/share/ocsinventory-reports/ocsreports/require/function_commun.php on line 103
[Thu Sep 27 15:32:45.110806 2018] [php7:warn] [pid 2730] [client 10.234.32.6:65164] PHP Warning:  count(): Parameter must be an array or an object that implements Countable in /usr/share/ocsinventory-reports/ocsreports/plugins/main_sections/ms_computer/ms_computer_views.php on line 128, referer: http://10.234.230.30/?function=visu_computers
[Thu Sep 27 15:34:05.496388 2018] [php7:warn] [pid 2726] [client 10.234.32.6:65168] PHP Warning:  count(): Parameter must be an array or an object that implements Countable in /usr/share/ocsinventory-reports/ocsreports/plugins/main_sections/ms_computer/ms_computer_views.php on line 128, referer: http://10.234.230.30/index.php?function=visu_computers
[Thu Sep 27 15:38:45.346696 2018] [php7:warn] [pid 2726] [client 10.234.32.6:65226] PHP Warning:  count(): Parameter must be an array or an object that implements Countable in /usr/share/ocsinventory-reports/ocsreports/plugins/main_sections/ms_computer/ms_computer_views.php on line 128, referer: http://10.234.230.30/index.php?function=visu_computers
[Thu Sep 27 15:40:23.427974 2018] [php7:warn] [pid 2725] [client 10.234.32.6:65236] PHP Warning:  vsprintf(): Too few arguments in /usr/share/ocsinventory-reports/ocsreports/require/function_commun.php on line 74, referer: http://10.234.230.30/index.php?function=officepack
[Thu Sep 27 15:40:23.428062 2018] [php7:warn] [pid 2725] [client 10.234.32.6:65236] PHP Warning:  mysqli_query(): Empty query in /usr/share/ocsinventory-reports/ocsreports/require/function_commun.php on line 103, referer: http://10.234.230.30/index.php?function=officepack
[Thu Sep 27 15:40:29.656913 2018] [php7:warn] [pid 2725] [client 10.234.32.6:65236] PHP Warning:  count(): Parameter must be an array or an object that implements Countable in /usr/share/ocsinventory-reports/ocsreports/plugins/main_sections/ms_computer/ms_computer_views.php on line 128, referer: http://10.234.230.30/index.php?function=visu_computers
[Thu Sep 27 15:45:21.306623 2018] [php7:warn] [pid 1882] [client 10.234.32.6:65250] PHP Warning:  count(): Parameter must be an array or an object that implements Countable in /usr/share/ocsinventory-reports/ocsreports/plugins/main_sections/ms_computer/ms_computer_views.php on line 128, referer: http://10.234.230.30/index.php?function=visu_computers
[Thu Sep 27 17:48:18.593565 2018] [php7:warn] [pid 2725] [client 10.234.242.84:50308] PHP Warning:  vsprintf(): Too few arguments in /usr/share/ocsinventory-reports/ocsreports/require/function_commun.php on line 74, referer: http://10.234.230.30/
[Thu Sep 27 17:48:18.593643 2018] [php7:warn] [pid 2725] [client 10.234.242.84:50308] PHP Warning:  mysqli_query(): Empty query in /usr/share/ocsinventory-reports/ocsreports/require/function_commun.php on line 103, referer: http://10.234.230.30/




tail -f /var/log/mysql/error.log
2018-09-27 13:41:26 140477911892544 [Note] InnoDB: Completed initialization of buffer pool
2018-09-27 13:41:26 140477911892544 [Note] InnoDB: Highest supported file format is Barracuda.
2018-09-27 13:41:26 140477911892544 [Note] InnoDB: 128 rollback segment(s) are active.
2018-09-27 13:41:26 140477911892544 [Note] InnoDB: Waiting for purge to start
2018-09-27 13:41:26 140477911892544 [Note] InnoDB:  Percona XtraDB (http://www.percona.com) 5.6.36-82.1 started; log sequence number 3064094
2018-09-27 13:41:26 140477261145856 [Note] InnoDB: Dumping buffer pool(s) not yet started
2018-09-27 13:41:26 140477911892544 [Note] Plugin 'FEEDBACK' is disabled.
2018-09-27 13:41:26 140477911892544 [Note] Server socket created on IP: '127.0.0.1'.
2018-09-27 13:41:26 140477911892544 [Note] /usr/sbin/mysqld: ready for connections.
Version: '10.1.26-MariaDB-0+deb9u1'  socket: '/var/run/mysqld/mysqld.sock'  port: 3306  Debian 9.1

**Comment by helicrenz** [comment]
@alexchini ,

Yes, i already check the permission its already www-data. 

-rw-r--r--  1 www-data www-data  34K Sep 27 15:55 officepack.zip
-rwxr-xr-x  1 www-data www-data  127 Sep 24 11:24 README

I just confused why https://github.com/PluginsOCSInventory-NG/officepack/releases officepack 2.0 doesnt work. While this one work https://github.com/PluginsOCSInventory-NG/officepack/issues/34 but its unable to show the data office keys licensed.

**Comment by helicrenz** [comment]
tail -f /var/log/apache2/error.log | grep office
[Thu Sep 27 15:40:23.427974 2018] [php7:warn] [pid 2725] [client 10.234.32.6:65236] PHP Warning:  vsprintf(): Too few arguments in /usr/share/ocsinventory-reports/ocsreports/require/function_commun.php on line 74, referer: http://10.234.230.30/index.php?function=officepack
[Thu Sep 27 15:40:23.428062 2018] [php7:warn] [pid 2725] [client 10.234.32.6:65236] PHP Warning:  mysqli_query(): Empty query in /usr/share/ocsinventory-reports/ocsreports/require/function_commun.php on line 103, referer: http://10.234.230.30/index.php?function=officepack
[Thu Sep 27 17:57:11.356767 2018] [php7:warn] [pid 2727] [client 10.234.242.84:50438] PHP Warning:  require(/usr/share/ocsinventory-reports/ocsreports/plugins/main_sections/ms_officepack/ms_officepack.php): failed to open stream: No such file or directory in /usr/share/ocsinventory-reports/ocsreports/require/header.php on line 437
[Thu Sep 27 17:57:11.356891 2018] [php7:error] [pid 2727] [client 10.234.242.84:50438] PHP Fatal error:  require(): Failed opening required '/usr/share/ocsinventory-reports/ocsreports/plugins/main_sections/ms_officepack/ms_officepack.php' (include_path='.:/usr/share/php') in /usr/share/ocsinventory-reports/ocsreports/require/header.php on line 437
[Thu Sep 27 18:00:35.470654 2018] [php7:warn] [pid 2730] [client 10.234.242.84:50454] PHP Warning:  require(/usr/share/ocsinventory-reports/ocsreports/plugins/main_sections/ms_officepack/ms_officepack.php): failed to open stream: No such file or directory in /usr/share/ocsinventory-reports/ocsreports/require/header.php on line 437
[Thu Sep 27 18:00:35.470746 2018] [php7:error] [pid 2730] [client 10.234.242.84:50454] PHP Fatal error:  require(): Failed opening required '/usr/share/ocsinventory-reports/ocsreports/plugins/main_sections/ms_officepack/ms_officepack.php' (include_path='.:/usr/share/php') in /usr/share/ocsinventory-reports/ocsreports/require/header.php on line 437

**Comment by helicrenz** [comment]
i manage to fix mine. how do you automate the vbs script using ocs package? any suggestion

**Comment by alexchini** [comment]
the vbs script is load in the client by the OCSInventory.exe  as you can see in the posts before. You can also manually run the vbs script in the client u should see some popup with the office license key, if you don't see nothing it mean the vbscript is not able to parse your office version

**Comment by helicrenz** [comment]
yes i manage to make it work. Do you think its possible to use OCS deployment to store the vbs script on each client of computer? or its better to use GPO?

**Comment by alexchini** [comment]
yes you can use the OCS deployment but you have to configure it in the right way, it is not possible tu run ocsinventory client as service, it must be scheduled in another way.

**Comment by helicrenz** [comment]
@alexchini really appreciate your feedback. i try to access http://192.168.1.30/download/ im getting 403 forbidden.  

I already set the right permission 

ls -lah /var/lib/ocsinventory-reports/download
total 52K
drwxrwxr-x 3 www-data www-data 4.0K Sep 28 12:08 .
drwxr-xr-x 7 www-data www-data 4.0K Sep 24 11:24 ..
drwxrwxrwx 2 www-data www-data 4.0K Sep 28 11:37 1538125954
-rwxrwxrwx 1 www-data www-data  34K Sep 27 13:21 officepack.zip
-rwxrwxrwx 1 www-data www-data    5 Sep 28 12:08 test.html

errorlog


[Fri Sep 28 12:59:41.264380 2018] [autoindex:error] [pid 13609] [client 10.234.32.6:51816] AH01276: Cannot serve directory /var/lib/ocsinventory-reports/download/: No matching DirectoryIndex (index.html,index.cgi,index.pl,index.php,index.xhtml,index.htm) found, and server-generated directory index forbidden by Options directive


Do you have any idea?

**Comment by gehasia** [comment]
@helicrenz can you check your /etc/apache2/conf-enabled/ocsinventory-reports.conf

You should have 

```
<Directory /usr/share/ocsinventory-reports/ocsreports>
    # By default, users can use console from everywhere
      <IfModule mod_authz_core.c>
        # Apache 2.4
        Require all granted
          </IfModule>
          <IfModule !mod_authz_core.c>
        Order deny,allow
        Allow from all
      </IfModule>
    Options Indexes FollowSymLinks
    DirectoryIndex index.php
    AllowOverride Options
```

The "Options Indexes FollowSymLinks" is important here

Then check your download directory block :

```
<Directory /var/lib/ocsinventory-reports/download>
   <IfModule mod_authz_core.c>
     # Apache 2.4
     Require all granted
   </IfModule>
   <IfModule !mod_authz_core.c>
     Order deny,allow
     Allow from all
   </IfModule>
</Directory>
Alias /download /var/lib/ocsinventory-reports/download
```

This directory should be seen as an Index on your browser (listing files inside)

**Comment by Lea9250** [comment]
Hi all,

Newer versions of the plugin have been released since this issue was opened. If you are still having issues with the most recent plugin and OCS versions, feel free to re open.

Best regards,
Léa
